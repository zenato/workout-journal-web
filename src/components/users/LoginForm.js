import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import { handleChangeInput } from 'lib/form';
import { Input, Button, ErrorMessage } from 'components/form';
import styles from './LoginForm.scss';

const cx = classNames.bind(styles);

class LoginForm extends Component {
  state = {
    error: null,
    username: '',
    password: '',
  };

  constructor(props) {
    super(props);

    this.handleChangeInput = handleChangeInput(this);
  }

  componentWillReceiveProps({ error }) {
    this.setState({ error });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.onSubmit(this.state);
  };

  render() {
    const { error } = this.state;
    return (
      <form onSubmit={this.handleSubmit} className={cx('login-form')}>
        <div className={cx('item')}>
          <label htmlFor="login-username">Username</label>
          <div className={cx('field')}>
            <Input
              id="login-username"
              type="text"
              name="username"
              value={this.state.username || ''}
              onChange={this.handleChangeInput}
            />
            <ErrorMessage error={error} name="username" />
          </div>
        </div>
        <div className={cx('item')}>
          <label htmlFor="login-password">Password</label>
          <div className={cx('field')}>
            <Input
              id="login-password"
              type="password"
              name="password"
              value={this.state.password || ''}
              onChange={this.handleChangeInput}
            />
            <ErrorMessage error={error} name="password" />
          </div>
        </div>

        {error && (
          <div className={cx('error')}>
            <span>Oops, An expected error seems to have occurred.</span>
          </div>
        )}

        <div className={cx('tool')}>
          <Button type="submit" value="Login" className="primary" />
        </div>
      </form>
    );
  }
}

LoginForm.propTypes = {
  error: PropTypes.object,
  onSubmit: PropTypes.func.isRequired,
};

export default LoginForm;
