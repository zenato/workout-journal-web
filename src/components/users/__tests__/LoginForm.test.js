import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../LoginForm';

describe('LoginForm', () => {
  const props = {
    hasError: false,
    handleSubmit: jest.fn(),
  };

  it('should render error if not empty `error` props.', () => {
    const component = shallow(<LoginForm {...props} hasError={true} />);
    expect(component).toMatchSnapshot();
  });

  it('should call `onSubmit` if submit form.', () => {
    const component = shallow(<LoginForm {...props} />).dive();
    component.find('form').simulate('submit');
    expect(props.handleSubmit).toBeCalled();
  });
});
