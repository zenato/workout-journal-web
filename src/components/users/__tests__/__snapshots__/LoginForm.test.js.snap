// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render error if not empty \`error\` props. 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <FormGroup
    label="Username"
  >
    <Field
      component={[Function]}
      name="username"
      type="text"
    />
  </FormGroup>
  <FormGroup
    label="Password"
  >
    <Field
      component={[Function]}
      name="password"
      type="password"
    />
  </FormGroup>
  <styled.div>
    <span>
      Oops, An expected error seems to have occurred.
    </span>
  </styled.div>
  <styled.div>
    <Button
      color="primary"
      type="submit"
      value="Login"
    />
  </styled.div>
</styled.form>
`;
